dx/cqube-spec-checker/wide-search/widesearch.ts

```diff
@@ -0,0 +1,65 @@
+"use strict";
+exports.__esModule = true;
+var fs = require("fs");
+var path = require("path");
+var csv = require("csv-parser");
+
+function searchCSVFiles(rootDir, searchText) {
+  // Function to recursively search directories
+  function searchDirectories(directory) {
+    var files = fs.readdirSync(directory);
+    for (var _i = 0, files_1 = files; _i < files_1.length; _i++) {
+      var file = files_1[_i];
+      var filePath = path.join(directory, file);
+      var fileStat = fs.statSync(filePath);
+      if (fileStat.isDirectory()) {
+        // Recursive call for subdirectories
+        searchDirectories(filePath);
+      } else if (file.endsWith('.csv')) {
+        // Read CSV file and search for the text
+        var fileContent = fs.readFileSync(filePath, 'utf8');
+        if (fileContent.includes(searchText)) {
+          console.log('Text found in file:', filePath);
+
+          // Manipulate the CSV file
+          manipulateCSV(filePath);
+        }
+      }
+    }
+  }
+
+  // Start the search from the root directory
+  searchDirectories(rootDir);
+}
+
+function manipulateCSV(filePath) {
```

Comment:
What is this function doing?

Reply:
The `searchDirectories` function recursively searches directories and their subdirectories. It checks if each file is a directory or a CSV file. If it's a directory, it calls itself recursively. If it's a CSV file and contains the specified text, it calls the `manipulateCSV` function.

The `manipulateCSV` function reads a CSV file, collects the rows, manipulates them (e.g., setting a specific column to a new value), and overwrites the original file with the modified rows.